cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java

```diff
@@ -55,15 +65,40 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
     super.create(namespaceMeta);
     // TODO: CDAP-1519: Create base directory for filesets under namespace home
     // create HBase namespace
-    tableUtil.createNamespaceIfNotExists(getAdmin(), namespaceMeta.getNamespaceId().toId());
+    String namespace = tableUtil.getHBaseNamespace(namespaceMeta);
+    if (!Strings.isNullOrEmpty(namespaceMeta.getConfig().getHbaseNamespace())) {
+      if (!tableUtil.hasNamespace(getAdmin(), namespace)) {
+        throw new IOException(String.format("Custom mapped HBase namespace doesn't exist %s for namespace %s",
+                                            namespace, namespaceMeta.getName()));
+      }
+    } else {
+      tableUtil.createNamespaceIfNotExists(getAdmin(), namespace);
+    }
   }
 
+  @SuppressWarnings("ConstantConditions")
   @Override
   public void delete(NamespaceId namespaceId) throws IOException, ExploreException, SQLException {
     // soft delete namespace directory from filesystem
     super.delete(namespaceId);
     // delete HBase namespace
-    tableUtil.deleteNamespaceIfExists(getAdmin(), namespaceId.toId());
+    NamespaceConfig namespaceConfig = null;
+    try {
+      namespaceQueryAdmin.get(namespaceId.toId()).getConfig();
+    } catch (Exception ex) {
+      throw new IOException("Could not fetch custom HBase mapping.", ex);
+    }
+
+    String namespace = tableUtil.getHBaseNamespace(namespaceId.toId());
+    if (Strings.isNullOrEmpty(namespaceConfig.getHbaseNamespace())) {
+      // delete HBase namespace
+      tableUtil.deleteNamespaceIfExists(getAdmin(), namespace);
+    } else {
+      // custom namespace mapping is set for HBase, hence don't do anything during delete since the lifecycle of the
+      // namespace will be managed by the user
+      LOG.warn("Custom HBase mapping {} was found while deleting namespace {}. Hence skipping deletion of " +
```

Comment:
I don't think this should be at the WARN level, since it's perfectly normal operation.